snap/snapcraft.yaml

```diff
@@ -235,10 +235,11 @@ parts:
 
   mysql:
     plugin: cmake
-    source: https://github.com/kyrofa/mysql-server.git
+    source: https://github.com/mysql/mysql-server
     source-type: git
-    source-branch: mysql-5.7.18_support_no_setpriority
-    after: [boost]
+    source-tag: mysql-5.7.18
+    prepare: git apply $SNAPCRAFT_STAGE/support-compile-time-disabling-of-setpriority.patch
```

Comment:
I'm afraid you've hit the limits of patching in `prepare` here: it doesn't work for all plugins, most notably those that do out-of-source builds, like cmake. You can see this in the CI results: you're not building a patching mysql. Some context can be found [in this PR](https://github.com/snapcore/snapcraft/pull/1414), but long story short, this isn't possible without some hackery for now. There is a `post-pull` [in the pipeline](https://forum.snapcraft.io/t/scripting-and-setting-metadata-in-snapcraft/1447), I suggest we hold off on this patching until we can use that new functionality.